employee-rostering-shared/src/main/java/org/optaweb/employeerostering/shared/contract/Contract.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.shared.contract;
+
+import javax.persistence.Entity;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.optaweb.employeerostering.shared.common.AbstractPersistable;
+
+@Entity
+@NamedQueries({
+        @NamedQuery(name = "Contract.findAll",
+                query = "select c from Contract c" +
+                        " where c.tenantId = :tenantId" +
+                        " order by LOWER(c.name)"),
+        @NamedQuery(name = "Contract.deleteForTenant",
+                query = "delete from Contract c where c.tenantId = :tenantId")
+})
+@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"tenantId", "name"}),
+        @UniqueConstraint(columnNames = {"id"})})
+public class Contract extends AbstractPersistable {
+
+    @NotNull
+    @Size(min = 1, max = 120)
+    @Pattern(regexp = "^(?!\\s).*(?<!\\s)$", message = "Name should not contain any leading or trailing whitespaces")
+    private String name;
+
+    // Can be null
+    private Integer maximumMinutesPerDay;
+
+    // Can be null
+    private Integer maximumMinutesPerWeek;
+
+    // Can be null
+    private Integer maximumMinutesPerMonth;
+
+    // Can be null
+    private Integer maximumMinutesPerYear;
+
+    @SuppressWarnings("unused")
+    public Contract() {
+    }
+
+    public Contract(Integer tenantId, String name) {
+        this(tenantId, name, null, null, null, null);
+    }
+
+    public Contract(Integer tenantId, String name, Integer maximumMinutesPerDay, Integer maximumMinutesPerWeek, Integer maximumMinutesPerMonth, Integer maximumMinutesPerYear) {
+        super(tenantId);
+        this.name = name;
+        this.maximumMinutesPerDay = maximumMinutesPerDay;
+        this.maximumMinutesPerWeek = maximumMinutesPerWeek;
+        this.maximumMinutesPerMonth = maximumMinutesPerMonth;
+        this.maximumMinutesPerYear = maximumMinutesPerYear;
+    }
+
+    @AssertTrue
```

Comment:
IIRc, you can do an @Min(0) on numbers with java bean validation instead of doing a positive check via @AssertTrue (which is less declarative).

See usage of @Min on https://www.baeldung.com/javax-validation

Reply:
Btw, all these have a natural @Max too:

```
@Min(0) @Max(60L * 60L * 24L)
private Integer maximumMinutesPerDay
```

Reply:
Does `@min` work for null values (https://www.baeldung.com/javax-validation did not mentioned how the constraint handle null cases)? `maximumMinutesPerX` can be null if we want to ignore the constraint.